File: Easy/3360_Stone_Removal_Game/Solution.java



public class Solution {
    public boolean canAliceWin(int n) {
        boolean Alice = true;
        int turn = 10;

        while (n >= turn) {
            n -= turn;
            turn--;
            Alice = !Alice;
        }

        return !Alice;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // test cases 1
        int n1 = 12;
        System.out.println(solution.canAliceWin(n1));
        
        // test cases 2
        int n2 = 1;
        System.out.println(solution.canAliceWin(n2));
        
    }
}

File: Easy/3360_Stone_Removal_Game/Solution.cpp
#include "../../devlibs/cpp/cpphelper.h"

using namespace std;


// NOTE:
// For LeetCode submission, copy only the `class Solution` part.


class Solution {
public:
    bool canAliceWin(int n) {
        bool Alice = true;
        int turn = 10;

        while (n >= turn) {
            n -= turn;
            turn--;
            Alice = !Alice;
        }

        return !Alice;
    }
};


int main(int argc, char* argv[]) {
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    Solution solution;

    // test cases 1
    cout << solution.canAliceWin(12) << endl;

    // test cases 2
    cout << solution.canAliceWin(1) << endl;

    return 0;
}